client/src/components/Catalog.tsx

```diff
@@ -22,13 +22,26 @@ export function Catalog({ text, cards, limit }: Props) {
         alt={card.title}
       />
     );
-    cardWrapper.push(singleCard);
+    cardArray.push(singleCard);
+  }
+  if (cardArray.length % 18 && cardArray.length < limit) {
```

Comment:
We call the `18` here a "magic number". That's because it is not clear where it came from, or its purpose. Also, it is used in several different places, so if you change your mind on its value, there are several places you have to modify, and it will be easy to miss one.

In general, we never hard-code magic numbers. We always put them in a `const` variable, often at the top of the file (or top of the function).